libfuse/file.go

```diff
@@ -5,24 +5,58 @@
 package libfuse
 
 import (
+	"sync"
+
 	"bazil.org/fuse"
 	"bazil.org/fuse/fs"
 	"github.com/keybase/kbfs/libkbfs"
 	"golang.org/x/net/context"
 )
 
 type eiCache struct {
-	ei  libkbfs.EntryInfo
-	rid string
+	ei    libkbfs.EntryInfo
+	reqID string
+}
+
+// eiCacheHolder caches the EntryInfo for a particular reqID. It's used for the
+// Attr call after Create. This should only be used for operations with same
+// reqID.
+type eiCacheHolder struct {
+	mu    sync.Mutex
+	cache *eiCache
+}
+
+func (c *eiCacheHolder) destroy() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.cache = nil
+}
+
+func (c *eiCacheHolder) getAndDestroy(reqID string) (ei *libkbfs.EntryInfo) {
```

Comment:
Maybe `getAndDestroyIfMatches`?  Or something similar?  Otherwise it's a little confusing when `Attr` calls this with a wrong request ID.